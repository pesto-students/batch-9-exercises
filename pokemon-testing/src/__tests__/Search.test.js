import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Search from '../components/Search';

describe('<Search />', () => {
  beforeAll(() => {
    const mockFn = jest.fn();
    const props = { onChange: mockFn };
    const wrapper = shallow(<Search {...props} />);
  })
  test('<input> compulsorily renders', () => {
    const list = wrapper.find('input');
    expect(list).toHaveLength(1);
  });
  test('matches the snapshot', () => {
    const wrapper = shallow(<Search />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('makes no changes to passed prop', () => {
    const mockFn = jest.fn();
    const props = { onChange: mockFn };
    const wrapper = shallow(<Search {...props} />);
    const functionPassedAsProp = wrapper.find('input').props().onChange;
    expect(functionPassedAsProp).toBe(mockFn);
  });
  test('does not call passed function', () => {
    const mockFn = jest.fn();
    const props = { onChange: mockFn };
    shallow(<Search {...props} />);
    expect(mockFn).not.toHaveBeenCalled();
  });
});
