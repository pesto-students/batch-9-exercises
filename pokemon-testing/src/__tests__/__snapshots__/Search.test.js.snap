// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> matches the snapshot 1`] = `
<input
  placeholder="Enter pokemon name..."
  type="text"
/>
`;
